sdk/identity/azure-identity/azure/identity/_credentials/cli_credential.py

```diff
@@ -0,0 +1,66 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import platform
+_USE_SHELL = platform.system() == 'Windows'
+
+import json
+import time
+from datetime import datetime
+
+from subprocess import check_output, CalledProcessError, STDOUT
+import six
+
+from azure.core.credentials import AccessToken
+from azure.core.exceptions import ClientAuthenticationError
+
+
+def _microseconds_parsed(timestamp):
+    _index = timestamp.index('.')
+    _microseconds = timestamp[_index + 1:]
+    return '.'.join([timestamp[:_index]] + ['{:06d}'.format(int(_microseconds))])
+
+
+class CliCredential(object):
+
+    _LINUX_CMD_NOT_FOUND = 'command not found'
+    _WINDOWS_CMD_NOT_FOUND = 'is not recognized as'
```

Comment:
To prevent from actually printing the output of a command, the combination of parameter `shell` and `command` for `subprocess.run()` (as well as other `subprocess` methods) differs on  Windows, Linux and mac. 
Also, shell exits 1 for both Azure Cli not logged in (when `shell` specified `True`) and command not recognized on Windows.